api/nodes/serializers.py

```diff
@@ -98,16 +128,57 @@ def update(self, instance, validated_data):
 class NodePointersSerializer(JSONAPISerializer):
 
     id = ser.CharField(read_only=True, source='_id')
-    node_id = ser.CharField(source='node._id')
-    title = ser.CharField(source='node.title')
+    node_id = ser.CharField(source='node._id', help_text='The ID of the node that this pointer points to')
+    title = ser.CharField(read_only=True, source='node.title', help_text='The title of the node that this pointer '
+                                                                         'points to')
 
     class Meta:
         type_ = 'pointers'
 
     links = LinksField({
-        'html': 'absolute_url',
+        'html': 'get_absolute_url',
+    })
+
+    def get_absolute_url(self, obj):
+        pointer_node = Node.load(obj.node._id)
+        return pointer_node.absolute_url
+
+    def create(self, validated_data):
+        request = self.context['request']
+        user = request.user
+        auth = Auth(user)
+        node = self.context['view'].get_node()
+        pointer_node = Node.load(validated_data['node']['_id'])
+        pointer = node.add_pointer(pointer_node, auth, save=True)
+        return pointer
+
+    def update(self, instance, validated_data):
+        pass
+
+
+class NodeFilesSerializer(JSONAPISerializer):
+
+    id = ser.CharField(read_only=True, source='_id')
+    provider = ser.CharField(read_only=True)
+    path = ser.CharField(read_only=True)
+    item_type = ser.CharField(read_only=True)
+    name = ser.CharField(read_only=True)
+    metadata = ser.DictField(read_only=True)
+
+    class Meta:
+        type_ = 'files'
+
+    links = LinksField({
+        'self': WaterbutlerLink(kwargs={'node_id': '<node_id>'}),
+        'self_methods': 'valid_self_link_methods',
```

Comment:
It's a little involved; I blame Steve. :) There's no documentation as such. @samanehsan could explain it today, or I can try to explain it on Tuesday.